calico_node/startup/startup.go

```diff
@@ -906,6 +918,69 @@ func ensureDefaultConfig(ctx context.Context, cfg *apiconfig.CalicoAPIConfig, c
 	return nil
 }
 
+// ensureMigrated ensures any data migration needed is done.
+func ensureMigrated(ctx context.Context, cfg *apiconfig.CalicoAPIConfig, c client.Interface) error {
+	_, err := c.ClusterInformation().Get(ctx, "default", options.GetOptions{})
+	if err == nil {
+		// Nothing to do ClusterInformation exists so data has already been migrated
+		return nil
+	} else if _, ok := err.(cerrors.ErrorResourceDoesNotExist); !ok {
+		// The error indicates a problem with accessing the resource
+		return err
+	} else {
+		_, cv1, err := clients.LoadClients("", "")
+		if err != nil {
+			return err
+		}
+		// Grab the version from the clientv1
+		v, err := getV2ClusterVersion(cv1)
+		if err != nil {
+			if _, ok := err.(cerrors.ErrorResourceDoesNotExist); ok {
+				return nil
```

Comment:
So, this case is either:
- They're running a version so old it doesn't container this data.
- Someone deleted the data manually.

I think we need to return an error along the lines of "Unable to determine cluster version, cannot perform migration"

Reply:
What about if the CRDs aren't present (because it isn't an upgrade).  How are we handling that?

Reply:
@robbrockbank Yeah I think that would be this case. @caseydavenport I do think we need to check for something else to know if it is the case that the cluster is that old (or someone deleted the data), what should we look at for that?

Reply:
blech, right.

I don't know that we _can_ tell. We might just need to assume this is the "fresh cluster" case for now.